lib/_007/Val.pm

```diff
@@ -205,10 +205,86 @@ class Val::Str does Val {
 ###     say(/"Bond"/.search("J. Bond"));        # --> `True`
 ###
 class Val::Regex does Val {
-    has Val::Str $.contents;
+    # note: a regex should probably keep its lexpad or something to resolve calls&identifiers
+    has $.contents;
 
     method quoted-Str {
-        "/" ~ $.contents.quoted-Str ~ "/"
+        #"/" ~ $.contents.quoted-Str ~ "/"
+        "/...a regex.../"
+    }
+
+    method match(Str $str, Bool $full) {
```

Comment:
I especially don't like this one. Especially considering how it interacts with backtracking.